### PR TITLE
Refactor result class

### PR DESCRIPTION
**Pull request by ajreynol**
This significantly refactors the internal result class.  Entailments and "which" field are deleted, "Sat" is renamed to "Status".  Moreover "TYPE_NONE" is made into a status.

Simplifies the usage of this class throughout the code.

It also changes the API Result method `isSatUnknown` to `isUnknown`.

### CODE REVIEW

**Review by aniemetz — COMMENTED** [review]
Would it make sense to have a `Result::isSat()` (and friends), similarly as we have on the API level, instead of `getStatus()`?

**Comment by ajreynol** [comment]
I think there are a few places where we use the enum itself (e.g. in linear arithmetic), so that it makes sense to be able to extract it.  We could also add `isSat` and friends, although its not strictly necessary.

File: src/api/java/io/github/cvc5/api/Result.java [hunk 1]
```diff
@@ -117,12 +117,11 @@ public boolean isUnsat()
    * Return true if query was a checkSat() or checkSatAssuming() query and
    * cvc5 was not able to determine (un)satisfiability.
    */
-  public boolean isSatUnknown()
-  {
-    return isSatUnknown(pointer);
+  public boolean isUnknown() {
+    return isUnknown(pointer);
```
Comment (aniemetz) [inline src/api/java/io/github/cvc5/api/Result.java:121]: Format

File: src/smt/solver_engine.cpp [hunk 2]
```diff
@@ -634,15 +634,6 @@ void SolverEngine::defineFunctionRec(Node func,
   defineFunctionsRec(funcs, formals_multi, formulas, global);
 }
 
-Result SolverEngine::quickCheck()
```
Comment (aniemetz) [inline src/smt/solver_engine.cpp:637]: Why is this not needed anymore?

File: src/smt/solver_engine.cpp [hunk 3]
```diff
@@ -832,16 +823,16 @@ std::vector<Node> SolverEngine::getUnsatAssumptions(void)
...
   SolverEngineScope smts(this);
   finishInit();
   d_state->doPendingPops();
   ensureWellFormedTerm(formula, "assertFormula");
-  return assertFormulaInternal(formula);
+  assertFormulaInternal(formula);
 }
 
-Result SolverEngine::assertFormulaInternal(const Node& formula)
+void SolverEngine::assertFormulaInternal(const Node& formula)
```
Comment (aniemetz) [inline src/smt/solver_engine.cpp:835]: Why do we not return a result here anymore?  Ditto with blockModel and blockModelValues.

File: src/smt/solver_engine.cpp (hunk 2 above)
Comment (ajreynol) [reply src/smt/solver_engine.cpp:637]: It actually did nothing but return a (deprecated) entailment result.  This should have been deleted a long time ago.

File: src/api/java/io/github/cvc5/api/Result.java (hunk 1 above)
Comment (ajreynol) [reply src/api/java/io/github/cvc5/api/Result.java:121]: This is java formatting apparently.

File: src/smt/solver_engine.cpp (hunk 3 above)
Comment (ajreynol) [reply src/smt/solver_engine.cpp:835]: It returned the result of quickCheck, which for the same reasons as mentioned above should just be void.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**